addons/mail/models/mail_channel.py

```diff
@@ -96,14 +98,43 @@ def _compute_is_chat(self):
         for record in self:
             record.is_chat = record.channel_type == 'chat'
 
-    def _compute_is_member(self):
-        memberships = self.env['mail.channel.partner'].sudo().search([
-            ('channel_id', 'in', self.ids),
-            ('partner_id', '=', self.env.user.partner_id.id),
+    @api.depends('channel_last_seen_partner_ids.partner_id')
+    def _compute_channel_partner_ids(self):
+        for channel in self:
+            channel.channel_partner_ids = channel.channel_last_seen_partner_ids.partner_id
```

Comment:
Well I don't really understand what "channel_last_seen_partner_ids" is, naming is confusing, so it makes reading the rest a bit hard since it all depends on that.

Reply:
The m2m between channel and partner has a pivot table which is also a model (channel.partner). `channel_last_seen_partner_ids` is a confusing name since forever (we'll change it later) to this pivot table.
It is a membership relation, with extra info (between a channel and a partner) on it (last seen message, etc).